pkg/kubectl/cmd/top_node.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/metricsutil"
+)
+
+// TopNodeOptions contains all the options for running the top-node cli command.
+type TopNodeOptions struct {
+	ResourceName string
+	Selector     string
```

Comment:
> I would add the validation before running the command with either a usage error or a warning that the label will be ignored (if both name and selector were provided) rather than merge these into one field or providing any other information on which of those was provided. How about that?

Yeah, if one is ignored, then having two separate fields and a validation method that complains when both are set is perfect.